itou/employee_record/models.py

```diff
@@ -324,14 +324,17 @@ def update_as_rejected(self, code, label, archive):
 
         self.save()
 
-    def update_as_processed(self, code, label, archive):
-        if not self.status == Status.SENT:
+    def update_as_processed(self, code, label, archive, *, as_duplicate=False):
+        if self.status != Status.SENT or (as_duplicate and code != self.ASP_DUPLICATE_ERROR_CODE):
             raise InvalidStatusError(self.ERROR_EMPLOYEE_RECORD_INVALID_STATE)
```

Comment:
```suggestion
        if self.status != Status.SENT:
            raise InvalidStatusError(self.ERROR_EMPLOYEE_RECORD_INVALID_STATE)
        if as_duplicate and code != self.ASP_DUPLICATE_ERROR_CODE):
            raise InvalidStatusError(un_autre_message)
```
parce que dans le second cas le problème n'est pas `La fiche salarié n'est pas dans l'état requis pour cette action` mais plutôt `vous appelez la fonction avec les mauvais paramètres` :sweat_smile:

Reply:
J'avais peur de changer l'exception et donc qu'elle soit pas gérée mais en fait on attrape jamais de `InvalidStatusError()` en dehors des tests donc on peux même utiliser une `ValueError()` en fait.